MonoGame.Framework/Platform/Native/Platform.Interop.cs

```diff
@@ -0,0 +1,299 @@
+// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Input;
+using System.Runtime.InteropServices;
+
+namespace MonoGame.Interop;
+
+internal class MGHandleAttribute : System.Attribute
+{
+}
+
+internal enum EventType : uint
+{
+    Quit,
+
+    WindowMoved,
+    WindowResized,
+    WindowGainedFocus,
+    WindowLostFocus,
+    WindowClose,
+
+    KeyDown,
+    KeyUp,
+    TextInput,
+
+    MouseMove,
+    MouseButtonDown,
+    MouseButtonUp,
+    MouseWheel,
+
+    ControllerAdded,
+    ControllerRemoved,
+    ControllerStateChange,
+
+    DropFile,
+    DropComplete,
+}
+
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_WindowEvent
+{
+    public nint Window;
+    public int Data1;
+    public int Data2;
+}
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_KeyEvent
+{
+    public nint Window;
+    public uint Character;
+    public Keys Key;
+}
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_MouseMoveEvent
+{
+    public nint Window;
+    public int X;
+    public int Y;
+}
+
+internal enum MouseButton
+{
+    Left,
+    Middle,
+    Right,
+    X1,
+    X2
+}
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_MouseButtonEvent
+{
+    public nint Window;
+    public MouseButton Button;
+}
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_MouseWheelEvent
+{
+    public nint Window;
+    public int Scroll;
+    public int ScrollH;
+}
+
+internal enum ControllerInput : int
```

Comment:
@mrhelmut @harry-cpp By having one input type instead of seperate Axis and Button ones we can send just one event and not have extra unused fields or multiple structures.